### DISCUSSION TITLE
Samsung LC34G55T volume control is not working

### DISCUSSION THREAD

**Discussion by Delitants** [MonitorControl/MonitorControl#1185] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Samsung LC34G55T volume control is not working. Period.

### Expected behavior

So volume to change

### Anything else?

There is no DDC settings in the monitor menu

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.6
- Mac model: Intel
- MonitorControl version: 4.0.2
- Monitor(s): Samsung LC34G55T
- Monitor Cable(s)/Connection(s): HDMI
- Apple Silicon/M1 (yes or no): no
```

**Comment by the0neyouseek** [comment]
Hi @Neolo 
Just reading quickly it doesn't seems like the LC34G55T has integrated speakers ? Do you wish to control you mac internal speakers ?

**Comment by Delitants** [comment]
It has audio jack output for external speakers. Volume is controlled by monitor button only.

**Comment by jperkin** [comment]
I'm seeing the same with my Samsung LF32TU87, though in this case I'm connected over Thunderbolt, and have an M1.

The display controls work ok, though I have to use "Software (gamma, forced)", otherwise it only starts dimming the screen at around 60%.  However the audio controls do nothing, even though the icon shows up and is adjusting as expected, the output volume through the speakers remains unchanged.

This is all with the latest 4.1.0.

**Comment by x-magic** [comment]
> The display controls work ok, though I have to use "Software (gamma, forced)", otherwise it only starts dimming the screen at around 60%. However the audio controls do nothing, even though the icon shows up and is adjusting as expected, the output volume through the speakers remains unchanged.

I'm using Dell U4320Q and have the same issue - if the display is connected via DP or USB-C alt mode directly, DDC works for both picture and sound. But if connected via a Thunderbolt dock, only software control works. DDC can still be detected however, just controls won't work anymore. 

Maybe there should be a separate issue with Thunderbolt dock users? I assume DDC simply doesn't work through a dock.

**Comment by txoof** [comment]
Similar issue using a Dell U2723QE over a display port connection. The monitor has no internal speakers.
Setup:

- M1 Mac Mini Monterey 12.5
- Display Port Connection to U2723QE
- Powered speakers connected via 2.5mm jack

Symptoms:

- Output volume does not follow adjustments made via keyboard or MonitorControl app.
- Display dimming works as expected
- Volume control overlay appears and works as expected, when using F10-F12
- Volume slider in menu bar appears and can be adjusted

I think the U2723QE may have some sort of dock internally as it has a KVM switching capability so this may be related to @x-magic 's problem.

**Reply by x-magic** [reply]
That's my guess, either. Your model comes with KVM and DP out. DP out is usually provided via a MST chip (to allow DP daisy chains) and that can be an issue to pass along DDC/CI signal.

**Comment by samugem** [comment]
I have same issue with my Samsung S27HG5x when using usb c dock. If I use only usb c-hdmi adapter all works perfectly.

**Comment by waydabber** [comment]
Many displays do not properly support volume control over DDC. If brightness control works properly (not only the bottom 50% which indicates that DDC control is not working at all, either because of the use of the built-in HDMI port or lack of display support) and volume control does not, then there is nothing to do on the app side, but the display's firmware is at fault.

**Comment by x-magic** [comment]
An update to this as I got myself a new ThunderBolt dock. Here's my new setup:

- Dell U4320Q (still the same monitor - DDC/CI via built-in USB-C connection always work)
- CalDigit Element Hub (Thunderbolt 4 dock with JHL8440/Goshen Ridge chipset)
- A bi-directional DP1.4a cable (the kind that also works when connecting DP output to USB-C monitor)

DDC/CI works just fine on this setup. My previous dock was using JHL7440 (TB3, Titan Ridge) chipset where DDC/CI doesn't work. 

I tried a normal DP1.2 to USB-C cable which DDC/CI also works. So I guess cable doesn't really matter. It could also have something to do with how the dock is designed rather than which chipset it uses I believe, as my previous dock is WAVLINK-branded, while the new one is from a much more recognised brand, and maybe the former simply doesn't connect I<sup>2</sup>C between chipset and DP. (or probably because of the built-in MST hub)

**Comment by timbyu** [comment]
I have two Dell UltraSharp U2723QE monitors. They are both connected to an Anker Thunderbolt 4 dock using USB-c cables that came with the monitors. The dock is connected to my MacBook Pro M2 Max using a thunderbolt 4 cable. The screen brightness is synchronized beautifully and adjusts both external as well as the laptop screen using the standard Mac brightness keys. (Thank you!!) 

But I am experiencing a very similar failure of sound controls. I have my speakers plugged into the headphone jack on the first Dell monitor. Sound comes from the Mac to the speakers with no problem. Without your software Mac just locks the volume to 100% and does not let me controls the sound with volume keys. With your software, using the volume control keys acts like it is changing the volume including the onscreen indicator but the actual volume does not change. I tried bypassing the Doc and plugged the monitor directly into the MacBook with identical results, the onscreen volume indicator changes as I press the volume keys on the keyboard but the actual volume does not change. Any suggestions for steps I can try on my end would be welcome.

**Reply by x-magic** [reply]
Try `m1ddc` command line tool to directly send requests to your monitor through DDC/CI and test if your setup would passthrough the volume control to the external monitor. That's how MC is controlling volume of external monitors. Also check the monitor OSD volume control while testing and see if it reflects the changes through DDC/CI. 

Also there's a chance that 3.5mm line-out port on monitors may not support volume control in the first place - especially if there's no volume control option in OSD in the first place. Try a USB audio interface or the audio jack on your TB dock (if there's one) instead, rather than through the monitor.

**Reply by timbyu** [reply]
I thought the 3.5mm line-out was part of the docking station built into the monitor. There is no volume control in the OSD so I think you are right that there is not much hope of getting the volume to adjust. I tried a USB audio interface and it works fine with the native keys. I guess I will have to use that. I bought this model of monitor with the expectation I could use it as a dock including network, audio, and be able to daisy chain a second monitor. It has proven to be problematic with Mac and can't use it to daisy chain or use it with audio. Live and learn. If those XDR monitors were not so expensive, I would have gone that route. Thanks for your help!

**Reply by x-magic** [reply]
@timbyu Yes, that's correct. Dell displays with daisy chains are utilising DP MST and is never been supported on Macs, while most, if not all, Dell monitors with no built-in speakers won't have OSD or DDC/CI volume control support. In fact most dedicated docks are also using DP MST for multiple display outputs instead.  I think the built-in MST hub is the reason my original TB3 dock (I mentioned in a post above) doesn't do DDC/CI properly.
